### PR TITLE
Soft shadows 

### PR DESCRIPTION
**Pull request by lilleyse**
For #2594 

I'm using a basic 3x3 filter right now, so the results are alright but nothing special. Also included in this PR is simplifying the shaders a bit for point lights that don't use cube maps.

### CODE REVIEW

**Comment by lilleyse** [comment]
pcf_off
pcf

File: Source/Scene/ShadowMap.js [hunk 1]
```diff
@@ -135,6 +136,7 @@ define([
         //>>includeEnd('debug');
 
         this.enabled = false;
+        this._softShadows = defaultValue(options.softShadows, false);
```
Comment (pjcozzi) [inline Source/Scene/ShadowMap.js]: Can we expose this as a settable option?  If it is too painful to make it dynamic, we should at least be able to set it through Scene/Widget/Viewer at construction time.

File: Source/Renderer/AutomaticUniforms.js [hunk 2]
```diff
@@ -1588,6 +1588,20 @@ define([
...
             }
+        }),
+
+        /**
+         * An automatic GLSL uniform representing the size of a shadow map texel in the [0,1] range.
+         *
+         * @alias czm_shadowMapTexelSize
+         * @glslUniform
+         */
+        czm_shadowMapTexelSize : new AutomaticUniform({
```
Comment (pjcozzi) [inline Source/Renderer/AutomaticUniforms.js]: "Size" is confusing.  Maybe call this "step size" or "delta"  For a blast from the past, see https://github.com/AnalyticalGraphicsInc/cesium/compare/depthPostFX

File: Source/Scene/ShadowMap.js (hunk 1 above)
Comment (lilleyse) [inline Source/Scene/ShadowMap.js]: It won't work dynamically now as all shadow receivers would need to recompile their shaders, but I can make it public for construction time.

**Comment by pjcozzi** [comment]
If you didn't already, can you please update the shadows roadmap to include looking into other sampling strategies?

**Comment by pjcozzi** [comment]
Looks good!  Just those comments.  @bagnell can merge when ready.

File: Source/Scene/ShadowMap.js (hunk 1 above)
Comment (pjcozzi) [inline Source/Scene/ShadowMap.js]: OK

**Comment by lilleyse** [comment]
Ready.

**Pull request merged**